File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.java
class Solution {
    private int calculate(String n1, String n2, String oper){
        int a = Integer.parseInt(n1);
        int b = Integer.parseInt(n2);
        int ret = switch(oper){
                case "+" -> a + b;
                case "-" -> a - b;
                default -> 0;
        };
        return ret;
    }
    public String[] solution(String[] quiz) {
        String[] ret = new String[quiz.length];
        for(int i = 0; i < quiz.length; i++){
            String[] arr = quiz[i].split(" ");
            if(calculate(arr[0], arr[2], arr[1]) == Integer.parseInt(arr[4])) ret[i] = "O";
            else ret[i] = "X";            
        }
        return ret;
    }
}
/*
1. 모든 문자열에 대해
2. 문자열 계산 체크 반복
3. 문자열배열에 저장
*/

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.py
def solution(quiz):
    ret = []
    for q in quiz:
        print(q.split(" "))
        x, oper, y, eq, z = q.split(" ")
        x, y, z = int(x), int(y), int(z)
        if oper == '+' and x+y == z:
            ret.append("O")
        elif oper == '-' and x - y == z:
            ret.append("O")
        else:
            ret.append("X")
    return ret
    
"""
수식의 올바름 여부

"""